cmd/cainjector/app/start.go

```diff
@@ -97,21 +99,7 @@ servers and webhook servers.`,
 	return cmd
 }
 
-func (o InjectorControllerOptions) RunInjectorController(stopCh <-chan struct{}) {
-	eitherStopCh := make(chan struct{})
-	go func() {
-		defer close(eitherStopCh)
-		o.runCertificateBasedInjector(stopCh)
-	}()
-	go func() {
-		defer close(eitherStopCh)
-		o.runSecretBasedInjector(stopCh)
-	}()
-
-	<-eitherStopCh
-}
-
-func (o InjectorControllerOptions) runCertificateBasedInjector(stopCh <-chan struct{}) {
+func (o InjectorControllerOptions) RunInjectorController(ctx context.Context) error {
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
```

Comment:
This brings us back to having a single `Manager`, which is used only for leader election, metrics and as a place holder for global client settings.